src/arviz_plots/plots/essplot.py

```diff
@@ -0,0 +1,589 @@
+        >>>     rug=True,
+        >>> )
+
+    Relative ESS can be plotted instead of absolute:
+
+    .. plot::
+        :context: close-figs
+
+        >>> pc = plot_ess(
+        >>>     non_centered,
+        >>>     coords={"school": ["Choate", "Deerfield", "Hotchkiss"]},
+        >>>     relative=True,
+        >>> )
+
+    We can also adjust the number of points:
+
+    .. plot::
+        :context: close-figs
+
+        >>> pc = plot_ess(
+        >>>     non_centered,
+        >>>     coords={"school": ["Choate", "Deerfield", "Hotchkiss"]},
+        >>>     n_points=10,
+        >>> )
+
+    .. minigallery:: plot_ess
+
+    """
+    # initial defaults
+    if sample_dims is None:
+        sample_dims = rcParams["data.sample_dims"]
+    if isinstance(sample_dims, str):
+        sample_dims = [sample_dims]
+
+    ylabel = "{}"
+
+    # mutable inputs
+    if plot_kwargs is None:
+        plot_kwargs = {}
+    if pc_kwargs is None:
+        pc_kwargs = {}
+    else:
+        pc_kwargs = pc_kwargs.copy()
+
+    if stats_kwargs is None:
+        stats_kwargs = {}
+
+    # processing dt/group/coords/filtering
+    distribution = process_group_variables_coords(
+        dt, group=group, var_names=var_names, filter_vars=filter_vars, coords=coords
+    )
+
+    # ensuring plot_kwargs['rug'] is not False
+    rug_kwargs = copy(plot_kwargs.get("rug", {}))
+    if rug_kwargs is False:
+        raise ValueError("plot_kwargs['rug'] can't be False, use rug=False to remove the rug")
+
+    if backend is None:
+        if plot_collection is None:
+            backend = rcParams["plot.backend"]
+        else:
+            backend = plot_collection.backend
+
+    plot_bknd = import_module(f".backend.{backend}", package="arviz_plots")
+
+    # set plot collection initialization defaults
+    if plot_collection is None:
+        pc_kwargs["plot_grid_kws"] = pc_kwargs.get("plot_grid_kws", {}).copy()
+        pc_kwargs["aes"] = pc_kwargs.get("aes", {}).copy()
+        pc_kwargs.setdefault("col_wrap", 5)
+        pc_kwargs.setdefault(
+            "cols",
+            ["__variable__"]
+            + [dim for dim in distribution.dims if dim not in {"model"}.union(sample_dims)],
+        )
+        if "chain" in distribution:
+            pc_kwargs["aes"].setdefault("overlay", ["chain"])
+        if "model" in distribution:
+            pc_kwargs["aes"].setdefault("color", ["model"])
+            n_models = distribution.sizes["model"]
+            x_diff = min(1 / n_points / 3, 1 / n_points * n_models / 10)
+            pc_kwargs.setdefault("x", np.linspace(-x_diff, x_diff, n_models))
+            pc_kwargs["aes"].setdefault("x", ["model"])
+        aux_dim_list = [dim for dim in pc_kwargs["cols"] if dim != "__variable__"]
+        figsize = pc_kwargs.get("plot_grid_kws", {}).get("figsize", None)
+        figsize_units = pc_kwargs.get("plot_grid_kws", {}).get("figsize_units", "inches")
+        if figsize is None:
+            coeff = 0.2
```

Comment:
All the coeff related part was plot_forest specific, here the variable isn't used anywhere